### ISSUE TITLE
Can't compile

### ISSUE THREAD

**Issue by davidmurray** [luke-jr/bfgminer#320]
I get the following errors when compiling under raspbian:

miner.c: In function ‘probe_device_thread’:
miner.c:10414:3: warning: implicit declaration of function ‘LL_FOREACH2’ [-Wimplicit-function-declaration]
miner.c:10414:31: error: ‘same_devid_next’ undeclared (first use in this function)
miner.c:10414:31: note: each undeclared identifier is reported only once for each function it appears in
miner.c:10415:3: error: expected ‘;’ before ‘{’ token
miner.c:10451:4: error: expected ‘;’ before ‘{’ token
miner.c:10468:4: error: expected ‘;’ before ‘{’ token
miner.c:10488:3: error: expected ‘;’ before ‘{’ token

Thanks

**Comment by luke-jr** [comment]
What version of uthash?

**Comment by davidmurray** [comment]
1.9.5-1

**Comment by luke-jr** [comment]
Try pulling, I backported some macros...

**Comment by davidmurray** [comment]
Alright, now I get these errors:

bfgminer-fpgautils.o: In function `vcom_lowl_probe_w
rapper':
/home/pi/Bitcoin/bfgminer/fpgautils.c:658: undefined
 reference to`lowl_usb_attach_kernel_driver'
bfgminer-fpgautils.o: In function `_serial_autodetec
t_found_cb':
/home/pi/Bitcoin/bfgminer/fpgautils.c:685: undefined
 reference to`lowl_usb_attach_kernel_driver'
collect2: ld returned 1 exit st

**Comment by luke-jr** [comment]
Current git doesn't support building with anything less than all the default drivers. I'll fix that before 3.7, but it'd be a pain in the meantime with my current work.

**Comment by pantaluna** [comment]
I am getting a similar error for Linux Compile:
bfgminer-fpgautils.o: In function `vcom_lowl_probe_wrapper':
/srv/bitcoin.bitfury/bfgminer/fpgautils.c:658: undefined reference to`lowl_usb_attach_kernel_driver'

bfgminer-fpgautils.o: In function `_serial_autodetect_found_cb':
/srv/bitcoin.bitfury/bfgminer/fpgautils.c:685: undefined reference to`lowl_usb_attach_kernel_driver'

> What do I have to do to make the compile working fine , e.g. your remark "current git doesn't support building with anything less than all the default drivers."

**Comment by pantaluna** [comment]
FYI This git commit (V3.5.2) a2d734e43e27d270bf13ebd8f335fe74c6acee8f works fine.

**Comment by achow101** [comment]
In addition to the errors above, when I compile I am also getting this:

bfgminer-miner.o:miner.c: undefined reference to 'llround'

**Comment by luke-jr** [comment]
The errors above should be fixed.

Regarding llround, please open a new issue and mention details (like platform, libc, etc)

**Comment by ner00** [comment]
@achow101 I was also getting " undefined reference to llround' " before installing libusb..
Something like this:
"
 CCLD   bfgminer.exe
bfgminer-miner.o:miner.c:(.text+0x1755d): undefined reference to 'llround'
bfgminer-fpgautils.o:fpgautils.c:(.text+0x2ab8): undefined reference to 'lowl_usb_attach_kernel_driver'
bfgminer-fpgautils.o:fpgautils.c:(.text+0x2bc4): undefined reference to 'lowl_usb_attach_kernel_driver'
"
